File: src/main/java/site/gachontable/gachontablebe/domain/auth/domain/AccessToken.java
package site.gachontable.gachontablebe.domain.auth.domain;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

public record AccessToken(String accessToken) {

    public static AccessToken from(String jsonResponseBody) {
        JsonObject response = JsonParser.parseString(jsonResponseBody).getAsJsonObject();
        String accessToken = response.get("access_token").getAsString();

        return new AccessToken(accessToken);
    }
}

File: src/main/java/site/gachontable/domain/auth/domain/AccessToken.kt
package site.gachontable.domain.auth.domain

import com.google.gson.JsonParser

data class AccessToken(
    val accessToken: String,
) {
    companion object {
        fun from(jsonResponseBody: String): AccessToken {
            val response = JsonParser.parseString(jsonResponseBody).getAsJsonObject()
            val accessToken = response.get("access_token").asString

            return AccessToken(accessToken)
        }
    }
}
